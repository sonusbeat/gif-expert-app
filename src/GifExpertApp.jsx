import { useState } from "react";
import "./GifExpertApp.scss";
import AddCategory from './components/AddCategory';
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  // const [ categories, setCategories ] = useState([]); // Originalmente
  const [ categories, setCategories ] = useState(defaultCategories);

  return (
    <div className="container">
      <h1>Gift Expert App</h1>

      <AddCategory setCategories={ setCategories } />

      <hr />

      {
        categories.map(category => (
          <GifGrid key={category} category={ category } />
        ))
      }

    </div>
  );
};

export default GifExpertApp;
