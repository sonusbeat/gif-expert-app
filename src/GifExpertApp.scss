h1, h2 {
  text-align: center;
}

h1 {
  margin: 40px 0 30px;
}

h2 {
  text-align: center;
  margin-bottom: 1.2rem;
}