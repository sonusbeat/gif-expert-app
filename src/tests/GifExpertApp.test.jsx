import GifExpertApp from '../GifExpertApp';
import { shallow } from 'enzyme';
import "@testing-library/jest-dom";

describe("Pruebas en <GifExpertApp />", () => {
   test("Debe coincidir con el snapshot", () => {
    const wrapper = shallow( <GifExpertApp /> );
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un titulo h1 con el nombre de la app", () => {
    const wrapper = shallow( <GifExpertApp /> );
    const text = "Gift Expert App";
    const h1 = wrapper.find("h1");

    expect(h1.text().trim()).toBe(text);
  });

  test("Debe de mostrar una lista de categorÃ­as", () => {
    const categories = ["Dragon Ball Super", "Saint Seiya"];

    const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find("GifGrid").length ).toBe( categories.length );
  });
});