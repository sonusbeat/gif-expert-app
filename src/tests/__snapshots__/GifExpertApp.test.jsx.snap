// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe coincidir con el snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Gift Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</div>
`;

exports[`Pruebas en <GifExpertApp /> Debe de mostrar una lista de categor√≠as 1`] = `
<div
  className="container"
>
  <h1>
    Gift Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Dragon Ball Super"
    key="Dragon Ball Super"
  />
  <GifGrid
    category="Saint Seiya"
    key="Saint Seiya"
  />
</div>
`;
